java-security/src/test/java/com/sap/cloud/security/adapter/spring/SpringXsuaaSecurityContextTest.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.commons.io.IOUtils;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.lang.Nullable;
+import org.springframework.security.oauth2.jwt.Jwt;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import static com.sap.cloud.security.config.Service.IAS;
+import static com.sap.cloud.security.token.TokenClaims.SUBJECT;
+import static com.sap.cloud.security.xsuaa.token.SpringSecurityContext.*;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+public class SpringXsuaaSecurityContextTest {
+	private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+	AccessToken token;
+	SapIdToken sapIdToken;
+
+	@Before
+	public void setUp() throws IOException {
+		token = new XsuaaToken(IOUtils.resourceToString("/xsuaaUserAccessTokenRSA256.txt", UTF_8));
+		sapIdToken = new SapIdToken(IOUtils.resourceToString("/iasOidcTokenRSA256.txt", UTF_8));
+		SpringSecurityContext.clear();
+	}
+
+	@Test
+	public void getToken_fromEmptySecurityContext_isNull() {
+		Token token = SpringSecurityContext.getToken();
+		assertThat(token).isNull();
+
+		token = SpringSecurityContext.getAccessToken();
+		assertThat(token).isNull();
+	}
+
+	@Test
+	public void getTokenReturnsAnyToken() {
+		// initialize SpringSecurityContext with provided token
+		setToken(sapIdToken);
+		assertThat(SpringSecurityContext.getToken()).isEqualTo(sapIdToken);
+
+		setToken(token);
+		assertThat(SpringSecurityContext.getToken()).isEqualTo(token);
+	}
+
+	@Test
+	public void getAccessTokenReturnsXsuaaToken() {
+		setToken(token);
+		assertThat(SpringSecurityContext.getAccessToken()).isEqualTo(token);
+	}
+
+	@Test
+	public void getTokenReturnsIasOidcToken() {
+		setToken(sapIdToken);
+		assertThat(SpringSecurityContext.getToken().getService()).isEqualTo(IAS);
+		assertThat(SpringSecurityContext.getToken().getClaimAsString(SUBJECT)).isEqualTo("P176945");
+	}
+
+	@Test
+	public void clear_removesToken() {
+		setToken(token);
+		SpringSecurityContext.clear();
+
+		assertThat(SpringSecurityContext.getToken()).isNull();
+	}
+
+	@Test
+	public void tokenNotAvailableInDifferentThread() throws ExecutionException, InterruptedException {
+		setToken(token);
+
+		Future<Token> tokenInOtherThread = executorService.submit(() -> SpringSecurityContext.getToken());
+
+		assertThat(tokenInOtherThread.get()).isNull();
+	}
+
+	@Test
+	public void clearingTokenInDifferentThreadDoesNotAffectMainThread()
+			throws ExecutionException, InterruptedException {
+		setToken(token);
+
+		executorService.submit(() -> SpringSecurityContext.clear()).get(); // run and await other thread
```

Comment:
you can shorten the lambda expression -> executorService.submit(SpringSecurityContext::clear).get()

Reply:
cool 🥇